lib/v2/oreno3d/main.js

```diff
@@ -14,81 +14,122 @@ const sortRename = {
-                        title: `${video_name} - ${authors}`,
-                        author: authors,
+                // 第二页数据分析+缓存
+                const sec_data = await get_sec_page_data(link);
+                // 传递
+                const raw_pic_link = sec_data.raw_pic_link;
+                const video_name = sec_data.video_name;
+                const authors = sec_data.authors;
+                const origins = sec_data.origins;
+                const characters = sec_data.characters;
+                const tags = sec_data.tags;
+                const desc = sec_data.desc;
+                const iwara_link = sec_data.iwara_link;
+                const oreno3d_link = sec_data.oreno3d_link;
+                // 打包,缓存HTML
+                const description = await art(path.join(__dirname, 'templates/description.art'), {
+                    raw_pic_link,
+                    video_name,
+                    authors,
+                    origins,
+                    characters,
+                    tags,
+                    desc,
+                    iwara_link,
+                    oreno3d_link
+                    });
+                const title = `${video_name} - ${authors}`;
+                const realData = await ctx.cache.tryGet(oreno3d_link, () => {
+                    const result = {
+                        title,
+                        author:authors,
                         link: oreno3d_link,
                         category: tags.split(' '),
-                        description: art(path.join(__dirname, 'templates/description.art'), {
-                            raw_pic_link,
-                            video_name,
-                            authors,
-                            origins,
-                            characters,
-                            tags,
-                            desc,
-                            iwara_link,
-                            oreno3d_link,
-                        }),
+                        description
                     };
+                    return result;
                 });
-            })
-            .get()
+                return realData;
+        }).get(),
     );
-
+    return {items, title};
+    }
+    // 参数定义
+    const sort = ctx.params.sort;
+    let pagelimit = ctx.params.pagelimit ?? 1;
+    // 根链接
+    const userUrl = get_user_url(rootUrl, ctx, sort);
+    // 确认最大页数并比较修改
+    const response = await got(userUrl);
+    const $ = cheerio.load(response.data);
+    const maxPageSelector = "div.container > main > div.g-main-list > ul.pagination > li:last-child > a";
+    // 页码判断
+    if ($(maxPageSelector)) {
+        const pageLink = new URLSearchParams($(maxPageSelector).attr('href'));
+        const actualNum = pageLink.get("page");// 获取最大页数
+        if (parseInt(pagelimit) >= parseInt(actualNum)) {pagelimit = actualNum;}
+        }
+    else {pagelimit = 1;}
+    // 构造页码数组
+    const linkList = [];
+    for (let i = 0; i < pagelimit; i++) {
```

Comment:
```suggestion
    for (let i = 1; i < pagelimit; i++) {
```
`${userUrl}` and `${userUrl}&page=1` are the same

Reply:
but it begins at ${userUrl}&page=1,and the default is 1

Reply:
${userUrl}&page=0 is invaild

Reply:
You are visiting `${userUrl}` and then `${userUrl}&page=1` which has the same content.
![image](https://user-images.githubusercontent.com/11386903/185422175-610a90cd-4bcf-47bf-a349-cb5b14048ae5.png)

Reply:
the first page which not has `&page` is use for check the real num of pages,so I do not know how to solve this promblem